hazelcast/src/main/java/com/hazelcast/wan/WanReplicationPublisher.java

```diff
@@ -49,14 +91,147 @@
     void republishReplicationEvent(WanReplicationEvent wanReplicationEvent);
 
     /**
-     * Checks the size of the WAN replication queue and throws an
-     * exception if it has been reached or crossed.
+     * Publishes a WAN anti-entropy event. This method may also process the
+     * event or trigger processing.
+     * NOTE: used only in Hazelcast Enterprise.
+     *
+     * @param event the WAN anti-entropy event
+     */
+    default void publishAntiEntropyEvent(WanAntiEntropyEvent event) {
+    }
+
+    /**
+     * Calls to this method will pause WAN event container polling. Effectively,
+     * pauses WAN replication for its {@link WanReplicationPublisher} instance.
+     * <p>
+     * WAN events will still be offered to WAN event containers but they won't
+     * be polled. This means that the containers might eventually fill up and start
+     * dropping events.
+     * <p>
+     * Calling this method on already paused {@link WanReplicationPublisher}
+     * instances will have no effect.
+     * <p></p>
+     * There is no synchronization with the thread polling the WAN event
+     * containers and trasmitting the events to the target cluster. This means
+     * that the containers may be polled even after this method returns.
+     * NOTE: used only in Hazelcast Enterprise.
+     *
+     * @see #resume()
+     * @see #stop()
+     */
+    default void pause() {
+    }
+
+    /**
+     * Calls to this method will stop WAN replication. In addition to not polling
+     * events as in the {@link #pause()} method, a publisher which is stopped
+     * should not accept new events. This method will not remove existing events.
+     * This means that once this method returns, there might still be some WAN
+     * events in the containers but these events will not be replicated until
+     * the publisher is resumed.
+     * <p>
+     * Calling this method on already stopped {@link WanReplicationPublisher}
+     * instances will have no effect.
+     * NOTE: used only in Hazelcast Enterprise.
+     *
+     * @see #resume()
+     * @see #pause()
+     */
+    default void stop() {
+    }
+
+    /**
+     * This method re-enables WAN event containers polling for a paused or stopped
+     * {@link WanReplicationPublisher} instance.
+     * <p>
+     * Calling this method on already running {@link WanReplicationPublisher}
+     * instances will have no effect.
+     * NOTE: used only in Hazelcast Enterprise.
+     *
+     * @see #pause()
+     * @see #stop()
+     */
+    default void resume() {
+    }
+
+    /**
+     * Gathers statistics of related {@link WanReplicationPublisher} instance.
+     * This method will always return the same instance.
+     * NOTE: used only in Hazelcast Enterprise.
+     *
+     * @return {@link LocalWanPublisherStats}
+     */
+    default LocalWanPublisherStats getStats() {
+        return null;
+    }
+
+    /**
+     * Removes all WAN events pending replication.
```

Comment:
Typo? --> `Removes all pending WAN replication events.`

Reply:
Actually it's correct, it should mean the same as "Removes all WAN events waiting for replication". But I'll change the wording to avoid confusion :)